subscriber.go

```diff
@@ -133,12 +133,18 @@ func WrapBlockHookWithSyncedCidTracker(cidsSeenSoFar *[]cid.Cid, blockHook Block
 // handler holds state that is specific to a peer
 type handler struct {
 	subscriber *Subscriber
-	latestSync ipld.Link
-	msgChan    chan cid.Cid
+	// syncMutex serializes the handling of individual syncs. This should only
+	// guard the actual handling of a sync, nothing else.
+	syncMutex sync.Mutex
+	// When grabbing this lock, make sure to encompass the syncMutex. It's
```

Comment:
maybe worth being super explicit here:
`this lock must be claimed before syncMutex to deadlocks`